corgi/tasks/prod_defs.py

```diff
@@ -63,192 +63,203 @@ def update_products() -> None:
+    for compose in composes:
+        composes_dict[compose["url"]] = compose["variants"]
+    name = pd_product_stream.pop("id")
+    if match_version := RE_VERSION_LIKE_STRING.search(name):
+        version = match_version.group()
+    logger.debug("Creating or updating Product Stream: name=%s", name)
+    product_stream, _ = ProductStream.objects.update_or_create(
+        name=name,
+        defaults={
+            "version": version,
+            "description": "",
+            "products": product,
+            "productversions": product_version,
+            "active": active,
+            "brew_tags": brew_tags_dict,
+            "meta_attr": pd_product_stream,
+            "yum_repositories": yum_repos,
+            "composes": composes_dict,
+        },
+    )
+    product_stream_node, _ = ProductNode.objects.get_or_create(
+        object_id=product_stream.pk,
+        defaults={
+            "parent": product_version_node,
+            "obj": product_stream,
+        },
+    )
+    parse_variants_from_brew_tags(brew_tags, name, product, product_stream, product_stream_node,
+                                  product_version)
+    parse_errata_info(errata_info, product, product_stream,
+                      product_stream_node, product_version)
+    product_stream.save()
+    # Save taxonomies for newly-created model at end of each loop iteration
+    # All child models + nodes have already been created and had taxonomies saved
+    # This should be a no-op since we link models directly upon creation above
+    # Keeping it here just to be safe, but we could remove in future
+    product_stream.save_product_taxonomy()
+    return version
 
-                    composes_dict = {}
-                    for compose in composes:
-                        composes_dict[compose["url"]] = compose["variants"]
 
-                    name = pd_product_stream.pop("id")
-                    if match_version := RE_VERSION_LIKE_STRING.search(name):
-                        version = match_version.group()
+def parse_variants_from_brew_tags(brew_tags, name, product, product_stream, product_stream_node, product_version):
+    # quay-3 Errata Tool product version Quay-3-RHEL-8 list too many brew tags
+    # Linking quay streams to the 8Base-Quay-3 variant here via brew tags leads
+    # to builds from later streams being included in earlier ones,
+    # see PROJQUAY-5312.
+    # The rhn_satellite_6 streams have brew tags, but those brew tags are associated
+    # with the RHEL-7-SATELLITE-6.10 ET Product Version. We skip them
+    # here to ensure the rhn_satelite_6.10 variants only get linked with that stream
+    # I haven't filed a product bug to get them fixed since 6.7 - 6.9 are no longer
+    # active. See CORGI-546 for more details.
+    if (
+            len(brew_tags) > 0
+            and product_version.name != "quay-3"
+            and name
+            not in ["rhn_satellite_6.7", "rhn_satellite_6.8", "rhn_satellite_6.9"]
+    ):
```

Comment:
Do we need to add satellite 6.10 or newer versions to this list? Or can we remove any of these / is e.g. satellite 6.7 EOL yet?

Also one FYI about style - you can avoid many levels of nesting like this, by writing the if-condition the other way around and returning early from it:
```suggestion
    if not brew_tags or product_version.name == "quay-3" or name in ("rhn_satellite_6.7", "rhn_satellite_6.8", "rhn_satellite_6.9")
        return
    # else do stuff as usual, but no need for an indented block / more levels of nesting
```
It makes it a little easier to read in very long / complex code, and SonarQube will probably rank the complexity of the code lower, but no need to switch if you prefer this form.

Reply:
No need to add satellite 6.10 or later as it uses a different way (errata_info I think) to define it's builds, not brew_tags.